src/query/plan/operator.cpp

```diff
@@ -1578,17 +1586,26 @@ class SingleSourceShortestPathCursor : public query::plan::Cursor {
             throw QueryRuntimeException("Expansion condition must evaluate to boolean or null.");
         }
       }
-      to_visit_next_.emplace_back(edge, vertex);
+      to_visit_next_.emplace_back(edge, vertex, curr_acc_path);
       processed_.emplace(vertex, edge);
     };
 
     // populates the to_visit_next_ structure with expansions
     // from the given vertex. skips expansions that don't satisfy
     // the "where" condition.
-    auto expand_from_vertex = [this, &expand_pair](const auto &vertex) {
+    auto expand_from_vertex = [this, &pull_memory, &frame, &expand_pair](const auto &vertex) {
       if (self_.common_.direction != EdgeAtom::Direction::IN) {
+        Path curr_acc_path(pull_memory);
+        if (self_.filter_lambda_.accumulated_path_symbol) {
+          curr_acc_path = frame[self_.filter_lambda_.accumulated_path_symbol.value()].ValuePath();
+        }
         auto out_edges = UnwrapEdgesResult(vertex.OutEdges(storage::View::OLD, self_.common_.edge_types)).edges;
-        for (const auto &edge : out_edges) expand_pair(edge, edge.To());
+        for (const auto &edge : out_edges) {
+          expand_pair(edge, edge.To());
+          if (self_.filter_lambda_.accumulated_path_symbol) {
+            frame[self_.filter_lambda_.accumulated_path_symbol.value()] = curr_acc_path;
```

Comment:
Why are you getting path on line 1600, and then copying it here, it shouldn't change in the meantime right?

Reply:
It's changed inside expand_pair. I restore the state of Path. But I can call Shrink() to remove last added part instead of copying full path. This will be better

Reply:
Yes, correct, Josip.